src/libMessage/Messenger.cpp

```diff
@@ -850,6 +850,25 @@ bool SerializeToArray(const T& protoMessage, vector<unsigned char>& dst,
                                        protoMessage.ByteSize());
 }
 
+template <class T>
+bool RepeatableToArray(const T& repeatable, vector<unsigned char>& dst,
+                       const unsigned int offset) {
+  int tempOffset = offset;
+  for (int i = 0; i < repeatable.size(); ++i) {
+    const int length_available = dst.size() - tempOffset;
+    const auto& element = repeatable.Get(i);
+    if (length_available < element.ByteSize()) {
```

Comment:
Since SerializeToArray already does the size check and resize, you don't need to do it here anymore

Reply:
Good point, i will update it.